src/openrct2/ride/Track.cpp

```diff
@@ -899,3 +899,104 @@ void TrackElement::SetHighlight(bool on)
     if (on)
         Flags2 |= TRACK_ELEMENT_FLAGS2_HIGHLIGHT;
 }
+
+bool TrackSupportsHasFlag(uint8_t queryFlag, uint8_t sequenceFlags)
+{
+    switch (queryFlag)
+    {
+        case TRACK_SUPPORT_HAS_SUPPORT:
+        case TRACK_SUPPORT_RCT1_EXTRA_SUPPORT:
+        case TRACK_SUPPORT_INVERSION_EXTRA_SUPPORT:
+        case TRACK_SUPPORT_COVERED_EXTRA_SUPPORT:
+        case TRACK_SUPPORT_SHOULD_HAVE_SUPPORT:
+            return sequenceFlags & queryFlag;
+        case TRACK_SUPPORT_LOOP_A:
+        case TRACK_SUPPORT_LOOP_B:
+        case TRACK_SUPPORT_LOOP_C:
+            return (sequenceFlags & TRACK_SUPPORT_CHAIRLIFT_SUPPORT) != TRACK_SUPPORT_CHAIRLIFT_SUPPORT
+                && sequenceFlags & queryFlag;
+        case TRACK_SUPPORT_CHAIRLIFT_SUPPORT:
+            return (sequenceFlags & TRACK_SUPPORT_CHAIRLIFT_SUPPORT) == TRACK_SUPPORT_CHAIRLIFT_SUPPORT;
+        default:
+            return false;
+    }
+}
+
+bool TrackSupportDefaultState(RideTypeDescriptor rtd, CoordsXY position, track_type_t trackType, uint8_t sequence)
+{
+    if (rtd.SupportsBehaviour & RideSupportsBehaviour::AlwaysHasSupports)
+    {
+        return true;
+    }
+    else
+    {
+        Guard::Assert(sequence < MaxSequencesPerPiece);
+        auto ted = GetTrackElementDescriptor(trackType);
+        // if it is only a chairlift, only give chairlift supports
+        if (rtd.SupportsBehaviour == RideSupportsBehaviour::ChairliftSupports)
+        {
+            return TrackSupportsHasFlag(TRACK_SUPPORT_CHAIRLIFT_SUPPORT, sequence);
```

Comment:
Special chairlift behavior in the off chance that one day users will have the option to create a chairlift support on flat, up25, and down 25 tracks.